src/python/pants/engine/target.py

```diff
@@ -703,7 +724,63 @@ def compute_value(
                     "either a single string or an iterable of strings (e.g. a list of strings)"
                 ),
             )
-        return tuple(sorted(str_list))
+        return tuple(str_list)
+
+
+class DictStringToStringField(PrimitiveField, metaclass=ABCMeta):
```

Comment:
The name is clunky, but I think that's okay. We don't really want to encourage people to use this regularly.

Reply:
In that case, how about adding a comment to that effect? So that we know to not look here first when we need something similar in a couple of months.

Reply:
> In that case, how about adding a comment to that effect? So that we know to not look here first when we need something similar in a couple of months.

Well, to clarify, we don't need to ban or strongly discourage this field. There is a time and place for a DictField, as evidenced by it being used multiple times in several distinct V1 backends.

In contrast, we do want to strongly discourage `StringOrStringSequenceField`, which we plan to deprecate and actively discourage:

https://github.com/pantsbuild/pants/blob/b6a84251b78e5a917582a88ef6fff8287ccc702d/src/python/pants/engine/target.py#L676-L706